text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,678 @@
+  and appropriate to provide. For instance, if the target can support dynamic
+  memory allocation, it must provide an implementation of `alloc` and the
+  associated data structures.
+- Building the target and running the testsuite for the target must not take
+  substantially longer than other targets.
+  - In particular, if building the target takes a reasonable amount of time,
+    but the target cannot run the testsuite in a timely fashion due to low
+    performance of either native code or accurate emulation, that alone may
+    prevent the target from qualifying as tier 1.
+- If running the testsuite requires additional infrastructure (such as physical
+  systems running the target), the target maintainers must arrange to provide
+  such resources to the Rust project, to the satisfaction and approval of the
+  Rust infrastructure team.
+  - Such resources may be provided via cloud systems, via emulation, or via
+    physical hardware.
+  - If the target requires the use of emulation to meet any of the tier
+    requirements, the approving teams for those requirements must have high
+    confidence in the accuracy of the emulation, such that discrepancies
+    between emulation and native operation that affect test results will
+    constitute a high-priority bug in either the emulation or the
+    implementation of the target.
+  - If it is not possible to run the target via emulation, these resources must
+    additionally be sufficient for the Rust infrastructure team to make them
+    available for access by Rust team members, for the purposes of development
+    and testing. (Note that the responsibility for doing target-specific
+    development to keep the target well maintained remains with the target
+    maintainers. This requirement ensures that it is possible for other
+    Rust developers to test the target, but does not obligate other Rust
+    developers to make target-specific fixes.)
+- New tier 1 targets must not have a hard requirement for signed, verified, or
+  otherwise "approved" binaries. Developers must be able to build, run, and
+  test binaries for the target on systems they control.  (Doing so may require
+  enabling some appropriate "developer mode" on such systems, but must not
+  require the payment of any additional fee or other consideration, or
+  agreement to any onerous legal agreements.)
+  - The Rust project may decide to supply appropriately signed binaries if
+    doing so provides a smoother experience for developers using the target,
+    and a tier 2 target with host tools already requires providing appropriate
+    mechanisms that enable our infrastructure to provide such signed binaries.
+    However, this additional tier 1 requirement ensures that Rust developers
+    can help develop and test Rust software for the target (including Rust
+    itself), and that development or testing for the target is not limited to
+    an exclusive group or entity.
```

Comment:
To clarify, is this intended to mean that tier 2 targets need not permit a non-exclusive group or entity to develop for this target? I find that surprising, particularly given that tier 2[^1] targets "must have value to people other than its maintainers. (It may still be a niche target, but it must not be exclusively useful for an inherently closed group.)"

[^1]: I would expect this for tier 3 targets, too, as I've noted on that section above.

Reply:
Tier 2 intentionally doesn't *require* that the target allow unsigned binaries; that's a tier 1 requirement. The intention is that a target with a hard requirement for signed/approved/etc binaries may be allowed at tier 2, but is not allowed at tier 1.

The target must still have value to people other than its maintainers, but my intention was that tier 2 targets *may* require developers to go through a process that allows them to sign binaries or get binaries approved or similar.

Reply:
I've also clarified the language here, to avoid having an explanation of motivation turn into an additional requirement itself.